tinygrad/helpers.py

```diff
@@ -82,6 +82,14 @@ def from_np(x) -> DType: return asdict(dtypes())[np.dtype(x).name]
   int64: Final[DType] = DType(2, 8, "int64", np.int64)
   uint8: Final[DType] = DType(0, 1, "uchar", np.uint8)
 
+class NPArray(np.ndarray):
```

Comment:
I see you implemented the wrapper class to permit adding a property, however, the `MetalBuffer` is what we need to keep alive, I tried this change to your branch on `ops_metal.py`, and it passed my [segfaulting test](https://github.com/geohot/tinygrad/issues/931) :) 
```
diff --git a/tinygrad/runtime/ops_metal.py b/tinygrad/runtime/ops_metal.py
index ccb0af04..d105ec1e 100644
--- a/tinygrad/runtime/ops_metal.py
+++ b/tinygrad/runtime/ops_metal.py
@@ -1,9 +1,10 @@
 # pip3 install pyobjc-framework-Metal pyobjc-framework-Cocoa pyobjc-framework-libdispatch
 import os, subprocess, pathlib
 import Metal, Cocoa, libdispatch # type: ignore
+import numpy as np
 from typing import List, Any
 from tinygrad.codegen.cstyle import CStyleCodegen, CStyleLanguage
-from tinygrad.helpers import prod, getenv, DEBUG, DType
+from tinygrad.helpers import prod, getenv, DEBUG, DType, NPArray
 from tinygrad.ops import Compiled
 from tinygrad.runtime.lib import RawBufferMapped
 
@@ -25,9 +26,9 @@ class RawMetalBuffer(RawBufferMapped):
   def __del__(self):
     self._buf.release()
     super().__del__()
-  def _buffer(self):
+  def toCPU(self) -> np.ndarray:
     METAL.synchronize()
-    return self._buf.contents().as_buffer(self._buf.length())
+    return NPArray(np.frombuffer(self._buf.contents().as_buffer(self._buf.length()), dtype=self.dtype.np),buffer=self)
 
 def unwrap(x):
   ret, err = x
```

Reply:
Thanks @kposborne2! This fixes most of the test fails on my intel Mac. I'm not sure why tests are still failing in CI though...

Reply:
There has to be a better way to do this. There's no way to register a backing buffer with numpy? I really don't like the subclass.